contents/tutorials/android-session-replays.md

```diff
@@ -0,0 +1,262 @@
+        super.onCreate()
+        val config = PostHogAndroidConfig(
+            apiKey = POSTHOG_API_KEY,
+            host = POSTHOG_HOST 
+        ).apply {
+            sessionReplay = true
+            sessionReplayConfig.maskAllTextInputs = false // Whether all texts are masked or redacted (default is enabled)
+            sessionReplayConfig.screenshot = true // required for JetPack Compose
+        }
+        PostHogAndroid.setup(this, config)
+    }
+}
+```
+
+To get your PostHog API key and host, [sign up to PostHog](https://us.posthog.com/signup). Then, you can find your API key and host in your [project settings](https://us.posthog.com/settings/project).
+
+We now need to register our custom application class. Go to `app/manifests/AndroidManifest.xml` and add `android:name=".MySessionReplaysApplication"` within the `<application>` tag:
+
+```XML file=app/manifests/AndroidManifest.xml
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools">
+    <!-- ... rest of the config -->
+    <application
+        android:name=".MySessionReplaysApplication"
+        <!-- ... rest of the config -->
+    </application>
+</manifest>
+```
+
+To check your setup, build and run your app a few times. Enter in any values in the textfields and click the **Log in** button. You should start session replays in the [Session replay tab](https://us.posthog.com/replay/recent) in PostHog 🎉
+
+<ProductScreenshot
+  imageLight={"https://res.cloudinary.com/dmukukwp6/image/upload/v1720608972/posthog.com/contents/Screenshot_2024-07-10_at_11.55.16_AM.png"} 
+  imageDark={"https://res.cloudinary.com/dmukukwp6/image/upload/v1720608972/posthog.com/contents/Screenshot_2024-07-10_at_11.55.29_AM.png"} 
+  alt="Android session replays in PostHog" 
+  classes="rounded"
+/>
+
+## 3. (Optional) Mask sensitive data
+
+Your replays may contain sensitive information. For example, if you're building a banking app you may not want to capture how much money a user has in their account.
+
+To replace any type of `View` with a redacted version in the replay, set the [tag](https://developer.android.com/reference/android/view/View#tags) to `ph-no-capture`.
+
+In the below code, we do this for the email and password fields:
+
+```kotlin file=MainActivity.kt
+// your existing imports
+import androidx.compose.ui.platform.testTag
+
+// your existing of your code
+
+@Composable
+fun WelcomeScreen(navController: NavController, modifier: Modifier = Modifier) {
+    Surface(modifier = Modifier.fillMaxSize()) {
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center
+        ) {
+            Text(
+                text = "Welcome!",
+                modifier = Modifier.testTag("ph-no-capture") // Add this modifier to blur the text in replays
+            )
```

Comment:
@marandaneto this approach didnt work for me. The Text still appears as unblurred in the replays. Any ideas on how to do this?

Reply:
mm I have to check, SwiftUI did require some extra stuff to make it work, I will check

Reply:
https://github.com/PostHog/posthog.com/pull/9049
a limitation sadly, will investigate deeper.

Reply:
Thanks! Is there an example of a component you can suggest I can use this on? Would love to show how to redact in this tutorial

Reply:
In this case, I'd not use Jetpack compose then until I figure this out.

Reply:
I see. Whats the alternative to Jetpack compose that i should use?

Reply:
just the standard declarative views using xml
https://developer.android.com/develop/ui/views/layout/declaring-layout